### ISSUE TITLE
TypeScript complex type formatting regression

### ISSUE THREAD

**Issue by pbomb** [prettier/prettier#2682]
This TypeScript type, which formatted like this with v1.5.3:

(before)
```js
type GetChatsSagaEffects =
  | CallEffect
  | PutEffect<
      | GetUsersRequestedAction
      | GetChatsSucceededAction
      | GetChatsFailedAction
      | GetChatsStartedAction
    >
  | SelectEffect
```
now formats like this with v1.6.0

(after)
```js
type GetChatsSagaEffects =
  | CallEffect
  | PutEffect<

        | GetUsersRequestedAction
        | GetChatsSucceededAction
        | GetChatsFailedAction
        | GetChatsStartedAction
    >
  | SelectEffect
```

Note that the JavaScript formatter for Flow does not have this regression. It continues to format like the first (before) example.

**Comment by azz** [comment]
Cannot seem to reproduce:

https://prettier.io/playground/#%7B%22content%22%3A%22%5Cntype%20GetChatsSagaEffects%20%3D%5Cn%20%20%7C%20CallEffect%5Cn%20%20%7C%20PutEffect%3C%5Cn%20%20%20%20%20%20%20%20%7C%20GetUsersRequestedAction%5Cn%20%20%20%20%20%20%20%20%7C%20GetChatsSucceededAction%5Cn%20%20%20%20%20%20%20%20%7C%20GetChatsFailedAction%5Cn%20%20%20%20%20%20%20%20%7C%20GetChatsStartedAction%5Cn%20%20%20%20%3E%5Cn%20%20%7C%20SelectEffect%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A4%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D

**Comment by vjeux** [comment]
The playground is still on 1.5.3 fyu

**Comment by pbomb** [comment]
I did repro it on the playground, actually, with [this link](https://prettier.io/playground/#%7B%22content%22%3A%22type%20GetChatsSagaEffects%20%3D%5Cn%20%20%7C%20CallEffect%5Cn%20%20%7C%20PutEffect%3C%5Cn%20%20%20%20%20%20%7C%20GetUsersRequestedAction%5Cn%20%20%20%20%20%20%7C%20GetChatsSucceededAction%5Cn%20%20%20%20%20%20%7C%20GetChatsFailedAction%5Cn%20%20%20%20%20%20%7C%20GetChatsStartedAction%5Cn%20%20%20%20%3E%5Cn%20%20%7C%20SelectEffect%22%2C%22options%22%3A%7B%22printWidth%22%3A100%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22es5%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Afalse%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D).

**Comment by vjeux** [comment]
@pbomb the playground meanwhile to be 1.6.0 :) It's indeed an issue and a regression.  @JamesHenry could you look into it, it's likely an issue with a node having being renamed and we didn't have tests to cover this edge case.

**Comment by JamesHenry** [comment]
Sure, will take a look tomorrow

**Comment by JamesHenry** [comment]
Ok, scratch that, I couldn't sleep 😄 https://github.com/prettier/prettier/pull/2688

**Comment by lydell** [comment]
(The playground showed up as 1.5.3 for me until I hit the "hard refresh" (no cache) shortcut – then 1.6.0 go there.)

**Comment by vjeux** [comment]
I just shipped 1.6.1 that contains this fix! Thanks for reporting it!

**Comment by pbomb** [comment]
Great, thanks!
